cardano-cli/src/Cardano/CLI/EraBased/Run/Governance/Actions.hs

```diff
@@ -121,10 +110,10 @@ runGovernanceActionInfoCmd
     firstExceptT GovernanceActionsCmdWriteFileError . newExceptT $
       conwayEraOnwardsConstraints eon $
         writeFileTextEnvelope outFile (Just "Info proposal") proposalProcedure
-   where
-    fetchURLErrorToGovernanceActionError
-      :: AnchorDataTypeCheck -> ExceptT FetchURLError IO a -> ExceptT GovernanceActionsError IO a
-    fetchURLErrorToGovernanceActionError adt = withExceptT (GovernanceActionsProposalFetchURLError adt)
+
+fetchURLErrorToGovernanceActionError
+  :: AnchorDataTypeCheck -> ExceptT FetchURLError IO a -> ExceptT GovernanceActionsError IO a
+fetchURLErrorToGovernanceActionError adt = withExceptT (GovernanceActionsProposalFetchURLError adt)
```

Comment:
You could get rid of the first argument by using something like this:
```haskell
data ProposalUrl
data ConstitutionUrl

class EntityName a where
  entityName :: Proxy a -> String

data TypeFirstWord a

instance Typeable a => EntityName (TypeFirstWord a) where
  entityName _ = maybe "" (fmap toLower . fst) . uncons . Data.List.Split.split splitter . show . someTypeRep $ Proxy @a
    where
      splitter = 
        Data.List.Split.dropBlanks
        . Data.List.Split.keepDelimsL
        $ Data.List.Split.whenElt isUpper

deriving via (TypeFirstWord ProposalUrl) instance EntityName ProposalUrl
deriving via (TypeFirstWord ConstitutionUrl) instance EntityName ConstitutionUrl
  

-- >>> entityName (Proxy @ProposalUrl)
-- "proposal"


-- >>> entityName (Proxy @ConstitutionUrl)
-- "constitution"
```
... or maybe it's a bit overkill for two types, but it was fun to write. 😄

Reply:
Hehe, interesting, I will add the idea to my toolbox.

But yes, I think it is a bit overkill here. Not least because it doesn't generalise well.
If I add `DRepMetadataUrl` for example, that will not work well